internal/resolution/manifest/maven.go

```diff
@@ -585,22 +629,7 @@ func write(buf *bytes.Buffer, w io.Writer, depPatches MavenDependencyPatches, pr
 					}
 					// Sort dependency management for consistency in testing.
 					sort.Slice(dm.Dependencies, func(i, j int) bool {
-						di := dm.Dependencies[i]
-						dj := dm.Dependencies[j]
-						if di.GroupID != dj.GroupID {
-							return di.GroupID < dj.GroupID
-						}
-						if di.ArtifactID != dj.ArtifactID {
-							return di.ArtifactID < dj.ArtifactID
-						}
-						if di.Type != dj.Type {
-							return di.Type < dj.Type
-						}
-						if di.Classifier != dj.Classifier {
-							return di.Classifier < dj.Classifier
-						}
-
-						return di.Version < dj.Version
+						return compareDependency(dm.Dependencies[i], dm.Dependencies[j])
```

Comment:
Not something to change, but something I've wondered about:
Is there a reason to use `sort.Slice` over [`slices.SortFunc`](https://pkg.go.dev/slices#SortFunc), or is it just a preference thing?

I guess technically `slices.SortFunc` would make this one line instead of two:
```golang
slices.SortFunc(dm.Dependencies, compareDependency)
```

Reply:
fwiw the docs technically recommend you do change it:

> Note: in many situations, the newer [slices.SortFunc](https://pkg.go.dev/slices#SortFunc) function is more ergonomic and runs faster.

though I think when I looked into it, it felt like their comment was more based on what types you were using that lead to different compare functions rather than "we've spent more time optimizing `slices.SortFunc` than this one" kind of thing?

also kind of related, here's [some prior art](https://github.com/google/osv-scanner/blob/main/internal/output/result.go#L23-L42) on how I implemented a chain of compares similar to what you've done with `compareDependency`.

Also also, note that there's [`cmp.Or`](https://pkg.go.dev/cmp@master#Or) which _might_ be useful here, but also might not be... 🤷

Reply:
Nice to know both `slice.SortFunc` and `com.Or`!

Reply:
sadly `cmp.Or` only for Go 1.22 :(

Reply:
Ek sorry I learned that the same hard way when I first discovered it 😬